arbos/l2pricing/l2pricing_constraints.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2025, Offchain Labs, Inc.
+// For license information, see https://github.com/OffchainLabs/nitro/blob/master/LICENSE.md
+
+package l2pricing
+
+import (
+	"github.com/ethereum/go-ethereum/arbitrum/multigas"
+
+	"github.com/offchainlabs/nitro/arbos/constraints"
+	"github.com/offchainlabs/nitro/arbos/storage"
+)
+
+type L2PricingStateConstraints struct {
+	L2PricingState
+	constraintsCount storage.StorageBackedUint64
+	resourceBacklogs [multigas.NumResourceKind]storage.StorageBackedUint64
+	constraints      constraints.ResourceConstraints
+}
+
+func InitializeL2PricingStateConstraints(sto *storage.Storage) error {
+	if err := InitializeL2PricingState(sto); err != nil {
+		return err
+	}
+
+	for resource := multigas.ResourceKindUnknown + 1; resource < multigas.NumResourceKind; resource++ {
+		offset := perTxGasLimitOffset + 1 + uint64(resource)
+		_ = sto.SetUint64ByUint64(offset, 0)
+	}
+
+	return nil
+}
+
+func OpenL2PricingStateConstraints(sto *storage.Storage, rc constraints.ResourceConstraints) *L2PricingStateConstraints {
```

Comment:
We should also get the constraints from storage, but here, there are only the backlogs. My initial idea was to store everything as JSON and use StorageBytes.
See https://github.com/OffchainLabs/nitro/pull/3461